core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -53,17 +55,47 @@ trait AutoTopicCreationManager {
     requestContext: RequestContext
   ): Unit
 
+  def getTopicCreationErrors(
+    topicNames: Set[String],
+    errorCacheTtlMs: Long
+  ): Map[String, String]
+
+  def close(): Unit = {}
+
 }
 
+case class CachedTopicCreationError(
```

Comment:
Is this public or private? Can we make this a purely internal thing inside the topic creation manager, since it does not appear in the interface?